### ISSUE TITLE
New rule suggestion: detect link that point to nothing

### ISSUE THREAD

**Issue by TravisEz13** [DavidAnson/markdownlint#144]
Example:

```markdown
[link][bad-link]

[good-link]: https://google.com
```

Currently this does not produce an error.

**Comment by stuartpb** [comment]
This should be broken into a couple of separate rules:

- No links that reference non-existent names (complaining about `[bad-link]`), which is unambiguously bad
- No named links with no existing references (complaining about `[good-link]`), which can be harmless

**Comment by TravisEz13** [comment]
@stuartpb yes, by intent was not to suggest the second rule, but I would love if it did.

**Comment by mpalumbo7** [comment]
I would like to see this rule added, too!

**Comment by lorenzleutgeb** [comment]
@DavidAnson I am thinking of implementing this (as two rules, like @stuartpb suggested). Could you please give indicate whether you would be willing to merge such rules (given that the implementation is solid and matches the requirements laid out in `CONTRIBUTING.md`)? I am asking in order to avoid useless work.

**Comment by DavidAnson** [comment]
@lorenzleutgeb This seems reasonable to do as two different rules. You might also see if this behavior makes sense to fold into one of them:￼ <https://github.com/DavidAnson/markdownlint/issues/213>. (I checked the other link-related issues and they do not seem relevant.)￼ If you'd like early and regular feedback, I'm happy to look over draft PR's as you make progress.￼ Thanks!
